File: blf-ts/src/blam/haloreach/v12065_11_08_24_1738_tu1actual/networking/online/files/online_file_summary_listing.ts
import { c } from "@craftycodie/cstruct";

@c.struct()
export class s_online_file_summary_listing_entry {
  @c.field("u64")
  share_id = 0n;

  @c.field("u32")
  screenshots_count = 0;

  @c.field("u32")
  films_count = 0;

  @c.field("u32")
  game_variants_count = 0;

  @c.field("u32")
  map_variants_count = 0;

  @c.field("u32")
  new_items_count = 0;

  @c.field("u32")
  unknown1C = 0;

  @c.field("u32")
  unknown20 = 0;
}

File: blf_lib/src/blam/haloreach/v12065_11_08_24_1738_tu1actual/networking/online/files/online_file_summary_listing.rs
use binrw::{BinRead, BinWrite};
use serde::{Deserialize, Serialize};

#[derive(Default, PartialEq, Debug, Clone, Serialize, Deserialize, BinRead, BinWrite)]
#[brw(big)]
pub struct s_online_file_summary_listing_entry {
    pub share_id: u64,
    pub screenshots_count: u32,
    pub films_count: u32,
    pub game_variants_count: u32,
    pub map_variants_count: u32,
    pub new_items_count: u32,
    pub unknown1C: u32,
    pub unknown20: u32,
}
